hana/lib/HANAService.js

```diff
@@ -30,6 +30,7 @@ class HANAService extends SQLService {
     this.on(['BEGIN'], this.onBEGIN)
     this.on(['COMMIT'], this.onCOMMIT)
     this.on(['ROLLBACK'], this.onROLLBACK)
+    this.on(['SELECT', 'INSERT', 'UPSERT', 'UPDATE', 'DELETE'], this.onNOTFOUND)
```

Comment:
```suggestion
    this.on('error', this.onNOTFOUND)
```

should we use the error event instead?

Reply:
> Also the handler should only be registered if multitenancy is enabled.

When the Database service is created it is not always configured as Multitenancy enabled. My approach would be to check whether the current context has a tenant defined or not.

Reply:
I've created a proposal for the error handler with this draft PR #597 
What I prefer with that implementation that it only kicks in if an error occurs. With the interceptor, there's a slight overhead for all successful requests as well.